### ISSUE TITLE
Failed to import `azurerm_cognitive_deployment`

### ISSUE THREAD

**Issue by magodo** [hashicorp/terraform-provider-azurerm#27166]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.0

### AzureRM Provider Version

4.0.0

### Affected Resource(s)/Data Source(s)

azurerm_cognitive_deployment

### Terraform Configuration Files

```hcl
N/A
```

### Debug Output/Panic Output

```shell
stderr=
  | 2024/08/23 16:12:59 profile: cpu profiling enabled, /tmp/profile1197563241/cpu.pprof
  | 2024-08-23T16:13:00.318+0800 [ERROR] terraform-provider-azurerm.terraform-provider-azurerm: [ERROR] setting state: Invalid address to set: []string{"scale"}
  | 2024-08-23T16:13:00.319+0800 [ERROR] terraform-provider-azurerm.terraform-provider-azurerm: Response contains error diagnostic: @caller=/home/magodo/github/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_summary="setting \"scale\": Invalid address to set: []string{\"scale\"}" tf_proto_version=5.6 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=0ba63195-7702-e1b9-e137-b5c84a43f33e diagnostic_detail="setting \"scale\": Invalid address to set: []string{\"scale\"}" diagnostic_severity=ERROR tf_resource_type=azurerm_cognitive_deployment tf_rpc=ReadResource timestamp="2024-08-23T16:13:00.319+0800"
  | 2024-08-23T16:13:00.321+0800 [ERROR] terraform-provider-azurerm: setting "scale": Invalid address to set: []string{"scale"}: setting "scale": Invalid address to set: []string{"scale"}
```

### Steps to Reproduce

`terraform import`

### Important Factoids

This seems to be some missing part when enabling 4.0

**Comment by fabiendelpierre** [comment]
I haven't tested this, but could it have to do with the change to resource provider registration?

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#improved-resource-provider-registration

The doc doesn't say clearly what the default behavior is but it sounds like "legacy" ([reference](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/resourceproviders/required.go#L133-L135)) may be the default. If it were me I'd start troubleshooting by adding this new `resource_provider_registrations` parameter and make sure Microsoft.CognitiveServices is included in the set of RPs you're using.

**Comment by JoshJarabek7** [comment]
I tried 'all' for provider registrations and it was still failing:

`Plan: 4 to add, 1 to change, 0 to destroy.
╷
│ Error: setting "scale": Invalid address to set: []string{"scale"}
│
│   with azurerm_cognitive_deployment.gpt_4o,
│   on main.tf line 306, in resource "azurerm_cognitive_deployment" "gpt_4o":
│  306: resource "azurerm_cognitive_deployment" "gpt_4o" {
│
│ setting "scale": Invalid address to set: []string{"scale"}
╵
`

However, upgrading from 4.0.0 to 4.0.1 seems to have fixed it. 👍

**Comment by magodo** [comment]
Fixed by #27173
